### PR TITLE
Fixes particle filter segfault 

### PR DESCRIPTION
**Pull request by moosahassanx**
Particle filter had a segfault looking at ball localisation

### CODE REVIEW

File: module/localisation/BallLocalisation/src/BallModel.hpp [hunk 1]
```diff
@@ -93,6 +96,10 @@ namespace module::localisation {
         Eigen::Matrix<Scalar, 2, 1> getRogueRange() const {
             return resetRange;
         }
+
+        int getParticleCount() {
```
Comment (KipHamiltons) [inline module/localisation/BallLocalisation/src/BallModel.hpp:100]:
```suggestion
        [[nodiscard]] int getParticleCount() const {
```

File: module/localisation/RobotParticleLocalisation/src/RobotModel.hpp [hunk 2]
```diff
@@ -144,6 +146,10 @@ namespace module::localisation {
             return resetRange;
         }
 
+        int getParticleCount() {
```
Comment (KipHamiltons) [inline module/localisation/RobotParticleLocalisation/src/RobotModel.hpp:149]:
```suggestion
        [[nodiscard]] int getParticleCount() const {
```

File: shared/tests/utility/math/filter/VanDerPolModel.hpp [hunk 3]
```diff
@@ -73,6 +75,10 @@ namespace shared::tests {
         StateVec getRogueRange() const {
             return reset_range;
         }
+
+        int getParticleCount() {
```
Comment (KipHamiltons) [inline shared/tests/utility/math/filter/VanDerPolModel.hpp:79]:
```suggestion
        [[nodiscard]] int getParticleCount() const {
```

**Review by KipHamiltons — CHANGES_REQUESTED** [review]

**Review by KipHamiltons — DISMISSED** [review]
LGTM

File: shared/tests/utility/math/filter/TestFilters.cpp [hunk 4]
```diff
@@ -245,12 +245,12 @@ TEST_CASE("Test the ParticleFilter", "[utility][math][filter][ParticleFilter]")
 
     INFO("Configuring the ParticleFilter with")
     INFO("    Time step..........: " << deltaT);
-    INFO("    Number of Particles: " << number_of_particles)
+    INFO("    Number of Particles: " << model_filter.model.getParticleCount());
```
Comment (Bidski) [inline shared/tests/utility/math/filter/TestFilters.cpp:248]: Need to set this in the model
https://github.com/NUbots/NUbots/blob/f03da89d6b52344603e8260032eb58284ea6e4dd/shared/tests/utility/math/filter/TestFilters.cpp#L168

**Review by Bidski — CHANGES_REQUESTED** [review]

**Review by Bidski — APPROVED** [review]
LGTM

**Pull request merged**